src/deserialize.cpp

```diff
@@ -0,0 +1,290 @@
+                }
+                case org::apache::arrow::flatbuf::Type::Binary:
+                    arrays.emplace_back(
+                        deserialize_non_owning_variable_size_binary<sparrow::binary_array>(
+                            record_batch,
+                            encapsulated_message.body(),
+                            name,
+                            metadata,
+                            buffer_index
+                        )
+                    );
+                    break;
+                case org::apache::arrow::flatbuf::Type::LargeBinary:
+                    arrays.emplace_back(
+                        deserialize_non_owning_variable_size_binary<sparrow::big_binary_array>(
+                            record_batch,
+                            encapsulated_message.body(),
+                            name,
+                            metadata,
+                            buffer_index
+                        )
+                    );
+                    break;
+                case org::apache::arrow::flatbuf::Type::Utf8:
+                    arrays.emplace_back(
+                        deserialize_non_owning_variable_size_binary<sparrow::string_array>(
+                            record_batch,
+                            encapsulated_message.body(),
+                            name,
+                            metadata,
+                            buffer_index
+                        )
+                    );
+                    break;
+                case org::apache::arrow::flatbuf::Type::LargeUtf8:
+                    arrays.emplace_back(
+                        deserialize_non_owning_variable_size_binary<sparrow::big_string_array>(
+                            record_batch,
+                            encapsulated_message.body(),
+                            name,
+                            metadata,
+                            buffer_index
+                        )
+                    );
+                    break;
+                default:
+                    throw std::runtime_error("Unsupported type.");
+            }
+        }
+        return arrays;
+    }
+
+    std::vector<sparrow::record_batch> deserialize_stream(std::span<const uint8_t> data)
+    {
+        const org::apache::arrow::flatbuf::Schema* schema = nullptr;
+        std::vector<sparrow::record_batch> record_batches;
+        std::vector<std::string_view> field_names;
+        std::vector<bool> fields_nullable;
+        std::vector<sparrow::data_type> field_types;
+        do
+        {
+            const auto [encapsulated_message, rest] = extract_encapsulated_message(data);
+            const org::apache::arrow::flatbuf::Message* message = encapsulated_message.flat_buffer_message();
+            switch (message->header_type())
+            {
+                case org::apache::arrow::flatbuf::MessageHeader::Schema:
+                {
+                    schema = message->header_as_Schema();
```

Comment:
Why do you have a function `deserialize_schema_message` if it's not being used here?